force-app/main/default/classes/WeatherController.cls

```diff
@@ -0,0 +1,33 @@
+public with sharing class WeatherController {
+    @AuraEnabled(cacheable=true)
+    public static WeatherWrapper getWeather(String lat, String lon) {
+
+        try{
+
+            String apiKey = 'a7b5f1c8f58b997bf986d8d2ebb64d81';
+            String apiUrl = 'https://api.openweathermap.org/data/2.5/weather?lat=' + lat + '&lon=' + lon + '&appid=' + apiKey + '&units=metric';
+
+            
+            Http http = new Http();
+            HttpRequest request = new HttpRequest();
+            request.setEndpoint(apiUrl);
+            request.setMethod('GET');
+            HttpResponse response = http.send(request);
+            
+            if(response.getStatusCode() == 200) {
+
+                WeatherWrapper weatherWrapper = (WeatherWrapper)JSON.deserialize(response.getBody(), WeatherWrapper.class);
+
+                system.debug(weatherWrapper.weather[0].description);
```

Comment:
no problem here, but in Production, always remove your system.debugs

You can use logs, create note objects, or whatever is best for you and the project, but never leave system.debugs in your code.